src/include/insn.h

```diff
@@ -102,11 +102,89 @@ NES_INSN_DEFN(asl) {
   }
 
   reg_block.update_z_flag(new_value);
-  if (new_value & 0b10000000) {
+  if (new_value & 0x80) {
     reg_block.update_n_flag(new_value);
   }
   reg_block.update_c_flag(old_value & 0x80);
   return true;
 }
 
+
+NES_INSN_DEFN(and) {
+  Reg8 old_value;
+  Reg8 new_value;
+  switch (opcode) {
+  case 0x29: { // Immediate
+    Data16 addr = mem.read_rom(reg_block.m_pc + 1, 1);
```

Comment:
Let's call the addr variable here operand or something, as it doesn't represent an addr